### PR TITLE
Add `$write_parquet()`

### LINKED ISSUE #745: is it possible to provide write_parquet in r-polars?
https://github.com/pola-rs/polars/issues/13092 sink_parquet has a major bug here; row_group_size is ignored. thanks!

### PR DESCRIPTION
**Pull request by ssh352**
please advise, thanks!

Close #745

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -91,6 +91,7 @@
         replacement of `pl$reset_options()`).
 -   New function `polars_envvars()` to print the list of environment variables
     related to polars (#735).
+-   New method `$write_parquet()` for DataFrame (#745).
```
Comment (etiennebacher) [inline NEWS.md:94]: Use the PR number instead of the issue number. You can also add your github tag after the PR number if you want

File: R/dataframe__frame.R [hunk 2]
```diff
@@ -1859,6 +1859,36 @@ DataFrame_write_csv = function(
     invisible()
 }
 
+#' @title  Write to parquet file
```
Comment (etiennebacher) [inline R/dataframe__frame.R:1862]: You can remove the `@title` tag

File: tests/testthat/test-parquet.R [hunk 3]
```diff
@@ -60,3 +60,17 @@ test_that("scan read parquet - parallel strategies", {
...
+  on.exit(unlink(tmpf))
+  df_exp = pl$DataFrame(mtcars)
+  df_exp$write_parquet(tmpf)
+
+  expect_identical(
+    pl$read_parquet(tmpf)$to_data_frame(),
+    mtcars,
+    ignore_attr = TRUE
+  )
+})
```
Comment (etiennebacher) [inline tests/testthat/test-parquet.R:76]: can you add a test to check that it throws an error if we pass a wrong `compression`?

File: R/dataframe__frame.R [hunk 4]
```diff
@@ -1859,6 +1859,36 @@ DataFrame_write_csv = function(
...
+#'
+#' @examples
+#' # write table 'mtcars' from mem to parquet
+#' dat = pl$DataFrame(mtcars)
+#'
+#' destination = tempfile(fileext = ".parquet")
+#' dat$write_parquet(destination)
+DataFrame_write_parquet = function(
+    path,
+    compression_method = "zstd",
```
Comment (etiennebacher) [inline R/dataframe__frame.R:1875]: This argument should be called `compression` instead (it can stay as `compression_method` on the Rust side as other functions in Rust use this name)

**Review by etiennebacher — CHANGES_REQUESTED** [review]
Thanks @ssh352, that looks good to me 👍 

I just have a few comments for you to address. Note that one cause of R CMD failure is that you need to update the snapshots. You can do that with `task test-snapshot-accept` (run `task` to see the list of all tasks available)

**Comment by ssh352** [comment]
running ` task test-all` generates a bunch diffs

I am not sure I should `task test-snapshot-accept` them

```
❯ gws
 M man/S3_as_arrow_table.Rd
 M man/S3_as_nanoarrow_array_stream.Rd
 M man/S3_as_record_batch_reader.Rd
 M man/S3_infer_nanoarrow_schema.Rd
 M man/S3_knit_print.Rd
?? tests/testthat/_snaps/after-wrappers.new.md
```

**Comment by etiennebacher** [comment]
Ah indeed the Taskfile should be updated. The thing is that we export methods for packages that we don't import by default (`knitr`, `nanoarrow`) so we need to load those packages when we re-document. I'll fix the Taskfile to do that automatically with `test-snapshot-accept`, but for now can you run `task build-documents` and then `testthat::snapshot_accept()`? That should remove the unwanted changes on Rd files

File: man/S3_as_arrow_table.Rd
Comment (eitsupi) [inline man/S3_as_arrow_table.Rd:1]: Unrelated change. Please do not commit these.
See https://github.com/pola-rs/r-polars/blob/a23e17d8e63f18a6502d03cc5f21de464d6d9069/DEVELOPMENT.md?plain=1#L71-L79

File: man/S3_as_nanoarrow_array_stream.Rd
Comment (eitsupi) [inline man/S3_as_nanoarrow_array_stream.Rd:1]: Ditto.

File: man/S3_as_record_batch_reader.Rd
Comment (eitsupi) [inline man/S3_as_record_batch_reader.Rd:1]: Ditto.

File: man/S3_infer_nanoarrow_schema.Rd
Comment (eitsupi) [inline man/S3_infer_nanoarrow_schema.Rd:1]: Ditto.

File: man/S3_knit_print.Rd
Comment (eitsupi) [inline man/S3_knit_print.Rd:1]: Ditto.

File: rustfmt.toml
Comment (eitsupi) [inline rustfmt.toml:1]: Could you open another PR to add this file and remove this here?

**Review by eitsupi — CHANGES_REQUESTED** [review]
Thanks for working on this!
Some minor comments for development.

**Comment by ssh352** [comment]
> now can you run `task build-documents` and then `testthat::snapshot_accept()`? That should remove the unwanted changes on Rd files

I ran the above

```
❯ gws
 M DESCRIPTION
 M man/S3_as_arrow_table.Rd
 M man/S3_as_nanoarrow_array_stream.Rd
 M man/S3_as_record_batch_reader.Rd
 M man/S3_infer_nanoarrow_schema.Rd
 M man/S3_knit_print.Rd
 M tests/testthat/_snaps/after-wrappers.md
 M tests/testthat/_snaps/dataframe.md
```
should I commit tests/testthat/_snaps/after-wrappers.md tests/testthat/_snaps/dataframe.md?

File: rustfmt.toml
Comment (ssh352) [reply rustfmt.toml:1]: ok

**Comment by eitsupi** [comment]
> should I commit tests/testthat/_snaps/after-wrappers.md tests/testthat/_snaps/dataframe.md?

Yes, please.

It seems that the version of rextendr you are using is incorrect for the DESCRIPTION file.
Other files probably need to be committed.

File: tests/testthat/_snaps/dataframe.md
Comment (eitsupi) [inline tests/testthat/_snaps/dataframe.md:1]: Sorry, I misunderstood. This update is ​​due to window size effects when running tests and should not be committed.

**Review by eitsupi — CHANGES_REQUESTED** [review]

**Review by eitsupi — CHANGES_REQUESTED** [review]
@ssh352 Could you revert unrelated Rd file changes?

**Comment by ssh352** [comment]
> @ssh352 Could you revert unrelated Rd file changes?

sorry is it because I clicked update branch with main?

should I revert to fdcee528ce85b2ed5b73b318a951ad8dde68b9ca?

**Comment by eitsupi** [comment]
@ssh352 Please check comments above.
https://github.com/pola-rs/r-polars/pull/758#issuecomment-1921166132
https://github.com/pola-rs/r-polars/pull/758#discussion_r1474344181

**Review by eitsupi — APPROVED** [review]
Thanks for your contribution!

**Review by etiennebacher — APPROVED** [review]
Thanks @ssh352
